web/src/engine/websites/ReaperScans.ts

```diff
@@ -48,66 +41,44 @@ export default class extends DecoratableMangaScraper {
-        return elements.map(element => new Chapter(this, manga, element.pathname, element.querySelector('div.truncate p.truncate').textContent.trim()));
+    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const { data } = await FetchJSON<APIResult<APIManga[]>>(new Request(new URL(`/query?page=${page}&perPage=100&series_type=Comic&adult=true`, this.apiUrl)));
+        return data.map(item => new Manga(this, provider, item.id.toString(), item.title.trim()));
     }
 
     public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
-        const uri = new URL(manga.Identifier, this.URI);
-        const [ data ] = await this.interactionTaskPool.Add(async () => FetchCSS<HTMLDivElement>(new Request(uri.href), 'main div[wire\\:id][wire\\:initial-data]'), Priority.Normal);
-        const chapterList = this.ExtractChapters(manga, data);
-        const body = JSON.parse(data.getAttribute('wire:initial-data')) as LaravelLivewireMessage;
-        delete body.effects;
-        for(let page = 2, run = true; run; page++) {
-            const chapters = await this.interactionTaskPool.Add(async () => this.FetchChaptersFromPage(manga, page, body), Priority.Normal);
+        const chapterList = [];
+        for (let page = 1, run = true; run; page++) {
+            const chapters = await this.interactionTaskPool.Add(async () => this.GetChaptersFromPage(manga, page), Priority.Normal);
             chapters.length > 0 ? chapterList.push(...chapters) : run = false;
         }
         return chapterList;
-    }
-
-    private async FetchChaptersFromPage(manga: Manga, page: number, body: LaravelLivewireMessage): Promise<Chapter[]> {
-        const uri = new URL(manga.Identifier, this.URI);
-        uri.pathname = '/livewire/message/' + body.fingerprint.name;
 
-        body.updates = [{
-            type: 'callMethod',
-            payload: {
-                id: '00000',
-                method: 'gotoPage',
-                params: [ page, 'page' ]
-            }
-        }];
-
-        const request = new Request(uri.href, {
-            method: 'POST',
-            body: JSON.stringify(body),
-            headers: {
-                'Content-Type': 'application/json'
-            }
-        });
-        const message = await FetchJSON<LaravelLivewireMessage>(request);
-        const dom = new DOMParser().parseFromString(message.effects.html, 'text/html');
-        return this.ExtractChapters(manga, dom);
+    }
+    private async GetChaptersFromPage(manga: Manga, page: number): Promise<Chapter[]> {
+        const { data } = await FetchJSON<APIResult<APIChapter[]>>(new Request(new URL(`/chapter/query?page=${page}&perPage=100&series_id=${manga.Identifier}`, this.apiUrl)));
+        return data.map(item => new Chapter(this, manga, `/series/${item.series.series_slug}/${item.chapter_slug}`, item.chapter_name.trim()));
     }
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
```

Comment:
Consider using the corresponding `PagesSinglePageCSS` instead.

Reply:
Website changed in the meantime : found invalid and proxied images, and i've adjusted CSS so FetchPages will have to stay.